src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestHandler.cs

```diff
@@ -0,0 +1,1417 @@
+
+            return header.ToString(Formatting.None);
+        }
+
+        /// <summary>
+        /// Creates a JSON representation of a HttpRequest payload.
+        /// </summary>
+        /// <param name="signedHttpRequestDescriptor">A structure that wraps parameters needed for SignedHttpRequest creation.</param>
+        /// <returns>A JSON representation of an HttpRequest payload.</returns>
+        /// <remarks>
+        /// Users can utilize <see cref="SignedHttpRequestCreationParameters.AdditionalClaimCreator"/> to create additional claim(s) and add them to the signed http request.
+        /// </remarks>
+        private protected virtual string CreateHttpRequestPayload(SignedHttpRequestDescriptor signedHttpRequestDescriptor)
+        {
+            Dictionary<string, object> payload = new Dictionary<string, object>();
+
+            AddAtClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateTs)
+                AddTsClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateM)
+                AddMClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateU)
+                AddUClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateP)
+                AddPClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateQ)
+                AddQClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateH)
+                AddHClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateB)
+                AddBClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateNonce)
+                AddNonceClaim(payload, signedHttpRequestDescriptor);
+
+            if (signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.CreateCnf)
+                AddCnfClaim(payload, signedHttpRequestDescriptor);
+
+            signedHttpRequestDescriptor.SignedHttpRequestCreationParameters.AdditionalClaimCreator?.Invoke(payload, signedHttpRequestDescriptor);
+
+            return ConvertToJson(payload);
+        }
+
+        /// <summary>
+        /// Converts a dictionary representation of HttpRequest into a JSON string.
+        /// </summary>
+        /// <param name="payload">HttpRequest payload represented as a <see cref="Dictionary{TKey, TValue}"/>.</param>
+        /// <returns>A JSON string.</returns>
+        protected virtual string ConvertToJson(Dictionary<string, object> payload)
```

Comment:
consider moving to utilities class

Reply:
I'll move it, it seems like the method fits better in the util class.

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/0fac2cc80a8875dea87c068ef99978ca51ad68ba

Reply:
Todo: Move to M.IM.Tokens assembly.